src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_Call.cs

```diff
@@ -820,6 +986,54 @@ private static ImmutableArray<RefKind> GetRefKindsOrNull(ArrayBuilder<RefKind> r
             }
         }
 
+        private static void AppendMissingOptionalArguments(SyntaxNode syntax,  
+            MethodSymbol optionalParametersMethod,
+            bool expanded, 
+            Binder binder,
+            ImmutableArray<ParameterSymbol> missingParameters,
+            ArrayBuilder<IArgument> argumentsBuilder)
+        {
+            ImmutableArray<ParameterSymbol> parametersOfOptionalParametersMethod = optionalParametersMethod.Parameters; 
+
+            foreach (ParameterSymbol parameter in missingParameters)
+            {
+                BoundExpression argument;
+                ArgumentKind kind;
+
+                // In case of indexer access, missing parameters are corresponding to the indexer symbol, we need to 
+                // get default values base on actual accessor method parameter symbols (but still want to tie resulted IArgument 
+                // to the indexer parameter.)
+                ParameterSymbol parameterOfOptionalParametersMethod = parametersOfOptionalParametersMethod[parameter.Ordinal];
+
+                if (expanded && parameterOfOptionalParametersMethod.Ordinal == parametersOfOptionalParametersMethod.Length - 1)
```

Comment:
>parametersOfOptionalParametersMethod.Length - 1 [](start = 79, length = 47)

I think this is not accurate, the length should come from a an indexer rather than from an accessor, because setter has an extra parameter that we shouldn't take into consideration. Please add a test.